const-oid/src/db.rs

```diff
@@ -82,6 +83,43 @@ impl<'a> Database<'a> {
 
         None
     }
+
+    /// Return the list of matched name for the OID.
+    pub const fn find_names_for_oid<'o>(&self, oid: &'o ObjectIdentifier) -> Names<'a, 'o> {
+        Names {
+            database: *self,
+            oid,
+            position: 0,
+        }
+    }
+}
+
+/// Iterator returning the multiple names that may be associated with an OID.
+pub struct Names<'a, 'o> {
+    database: Database<'a>,
+    oid: &'o ObjectIdentifier,
```

Comment:
`ObjectIdentifier` is `Copy`. You shouldn't need to borrow it and therefore have the extra lifetime.